### DISCUSSION TITLE
Not sure how to use a blob instead of a URL and open it in AR using scene-viewer or model-viewer?

### DISCUSSION THREAD

**Discussion by vasium** [google/model-viewer#2230] [Q&A]
I am trying to generate a glTF file from a custom 3D glTF configurator I am building. The file is being generated and I can download it without a problem if I use link.href = URL.createObjectURL( blob ); and link.download = filename; in the save(blob) function. However, what I am trying to do is to first generate the glTF file and then open it in AR using scene-viewer or model-viewer. Unfortunately when I am calling the exportGLTF() function the browser crashes! Any ideas of what I am doing wrong?

```
 var link = document.createElement('a');
  link.style.display = 'none';

  function exportGLTF() {
    var gltfExporter = new GLTFExporter();

    gltfExporter.parse(
      scene,
      function (result) {
        var output = JSON.stringify(result, null, 2);
        saveString(output, 'scene.gltf');
      },
      options
    );
  }

  function saveString(text, filename) {
    save(new Blob([text], { type: 'application/json' }), filename);
  }

  function save(blob) {
    link.href =
      'intent://arvr.google.com/scene-viewer/1.1?file=' +
      URL.createObjectURL(blob) +
      '&mode=ar_only#Intent;scheme=https;package=com.google.ar.core;action=android.intent.action.VIEW;end';
    link.click();
  }
```

**Comment by elalish** [comment]
I don't believe it's possible to open scene-viewer using a blob; @tpsiaki for confirmation. model-viewer also does not currently support this.

**Comment by vasium** [comment]
@elalish @tpsiaki Any idea how can I accomplish this with scene-viewer without using a blob?

**Comment by elalish** [comment]
@vasium If your configurator could be powered by model-viewer's scene-graph API, then it would work in webXR as well. Not sure there's much hope for scene-viewer unless you do something server-side. If you're interested in collaborating on expanding our scene-graph API, help would be most appreciated!

**Comment by d3tr0id** [comment]
Is there any news related to this? I have the same use case and have no solution :(

**Reply by elalish** [reply]
Use WebXR! It works perfectly. That's why it's our default AR mode.

**Comment by andrewvarga** [comment]
Just to confirm this is still the case, if I want to use the html anchor tag with the AR intent: `<a href="intent://arvr.google.com/scene-viewer/1.0?file=..."` the file can only be an absolute url, not something generated during runtime and resulting in a blob url or base64 encoded url? My tests show this but I want to confirm.
In this case what would be the best way to work around that? Either saving the file somewhere or using WebXR / embed model-viewer but then I won't be able to use the native android AR viewer I guess..

**Reply by elalish** [reply]
Correct. WebXR is the way - is there anything you find lacking compared to the native viewer? WebXR is faster, more consistent, and gives your page lots of customization options not available in the native viewer.

**Reply by cka29** [reply]
The 3D viewer would work out of the box if we could pass the blob to the scene viewer. For now I will use three.js for 3D viewing for android, because iOS supports blobs for quick-look (AR und 3D)

**Reply by elalish** [reply]
Hmm, but our default WebXR mode works on all the same Android devices that Scene Viewer does, handles Blobs, and works better. Why not just use that?
